rageval/metrics/_answer_bleu.py

```diff
@@ -0,0 +1,166 @@
+import re
+
+from rageval.metrics import Metric, add_attribute
+
+
+_DESCRIPTION = """\
+BLEU (Bilingual Evaluation Understudy) is an algorithm for evaluating the quality of text which has been machine-translated from one natural language to another.
+Scores are calculated by comparing individual translated segments, e.g., sentences, with a set of high-quality reference translations.
+Those scores are then averaged over the whole corpus to reach an estimate of the translation's overall quality.
+Neither intelligibility nor grammatical correctness are not taken into account.
+
+For details, see the paper: http://www.aclweb.org/anthology/P02-1040.pdf
+"""
+
+_KWARGS_DESCRIPTION = """\
+Args:
+    name : str
+    batch_size : int, Batch size for openai completion.
+
+Optional Args:
+    None
+
+Functions:
+    _clean: clean special word in sentence.
+    _compute_single: compute bleu score for single prediction with its references
+
+Examples:
+    >>> from datasets import Dataset
+    >>> import rageval as rl
+    >>> sample = {
+    ...     "answers": [
+    ...         "It is a guide to action which ensures that the military always obeys the commands of the party.",
+    ...         "It is to insure the troops forever hearing the activity guidebook that party direct.",
+    ...     ],
+    ...     "gt_answers": [
+    ...         [
+    ...             "It is a guide to action that ensures that the military will forever heed Party commands.",
+    ...             "It is the guiding principle which guarantees the military forces always being under the command of the Party.",
+    ...             "It is the practical guide for the army always to heed the directions of the party.",
+    ...         ],
+    ...         [
+    ...             "It is a guide to action that ensures that the military will forever heed Party commands.",
+    ...             "It is the guiding principle which guarantees the military forces always being under the command of the Party.",
+    ...             "It is the practical guide for the army always to heed the directions of the party.",
+    ...         ]
+    ...     ],
+    ... }
+    >>> dataset = Dataset.from_dict(sample)
+    >>> metric = rl.metrics.AnswerBleuScore()
+    >>> metric.mtype
+    'AnswerCorrectness'
+    >>> s, ds = metric.compute(dataset, batch_size=1)
+    >>> assert s == 0.3172992057845065
+    >>> type(ds)
+    <class 'datasets.arrow_dataset.Dataset'>
+
+"""
+
+
+_CITATION = """\
+@misc{Kishore2002bleu,
+      title={Bleu: a method for automatic evaluation of machine translation},
+      author={Kishore Papineni and Salim Roukos and Todd Ward and Wei-Jing Zhu},
+      year={2002},
+      page={311-318},
+      primaryClass={cs.CL}
+}
+"""
+
+
+@dataclass
+@add_attribute('mtype', 'AnswerCorrectness')
+@datasets.utils.file_utils.add_start_docstrings(_DESCRIPTION, _KWARGS_DESCRIPTION)
+class AnswerBleuScore(Metric):
+    """Bleu score computing with good quality reference."""
```

Comment:
please add notation that this metric can only evaluate English sentences but not Chinese, since it split sentences by space.